source/common/stream_info/utility.cc

```diff
@@ -43,6 +47,225 @@ absl::optional<ResponseFlag> ResponseFlagUtils::toResponseFlag(absl::string_view
   return absl::nullopt;
 }
 
+const absl::optional<uint32_t>
+ProxyStatusUtils::recommendedHttpStatusCode(const ProxyStatusError proxy_status) {
```

Comment:
Please add a link to the RFC in a comment to this method

Reply:
Done.